### PR TITLE
feat(daemon): wire migrate command to cli

### LINKED ISSUE #302: [P1] Wire migrate command to CLI
## Summary
The migration engine exists (9 modes, sophisticated scoring) but is dead code — `migrate_cli.rs` has `#![allow(dead_code)]` and the `Migrate` variant is not in the `Commands` enum. Wire it up.

**Epic**: #294
**Priority**: P1
**Effort**: S
**Week**: 1-2

## Acceptance Criteria
- [ ] Add `Migrate` variant to `Commands` enum in `crates/daemon/src/lib.rs`
- [ ] Wire dispatch in `main.rs`
- [ ] Remove `#![allow(dead_code)]` from `migrate_cli.rs`
- [ ] `loongclaw migrate --help` works
- [ ] Smoke test: `loongclaw migrate --mode discover --input .` runs without panic

## How to Claim
Comment with your GitHub handle to claim this issue.

## References
- Epic: #294
- Refs: #224 (unify migration surfaces)

### PR DESCRIPTION
**Pull request by finchxxia**
## Summary

- Problem: The migration CLI implementation already exists in `crates/daemon/src/migrate_cli.rs`, but it was not wired into the daemon’s top-level CLI surface, so `loongclaw migrate` was effectively dead code.
- Why it matters: This blocks the intended migration/operator flow, makes the existing 9-mode migration engine inaccessible from the shipped CLI, and leaves the repo out of sync with issue #302
- What changed: Added a top-level `Migrate` command to the daemon CLI, wired dispatch in `main.rs`, removed the `dead_code` allowance from `migrate_cli.rs`, and added CLI parse/help coverage to prove `loongclaw migrate --help` and representative mode parsing work.
- What did not change (scope boundary): This PR does not change migration engine behavior, scoring/selection logic, manifest/rollback behavior, onboarding/import flows, or app/spec tool surfaces such as `claw.migrate`.

## Linked Issues

- Closes #302

## Change Type

- [ ] Bug fix
- [x] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Security hardening
- [ ] CI / workflow / release

## Touched Areas

- [ ] Kernel / policy / approvals
- [ ] Contracts / protocol / spec
- [x] Daemon / CLI / install
- [ ] Providers / routing
- [ ] Tools
- [ ] Browser automation
- [ ] Channels / integrations
- [ ] ACP / conversation / session runtime
- [ ] Memory / context assembly
- [x] Config / migration / onboarding
- [ ] Docs / contributor workflow
- [ ] CI / release / workflows

## Risk Track

- [x] Track A (routine / low-risk)
- [ ] Track B (higher-risk / policy-impacting)

## Validation

- [x] `cargo fmt --all -- --check`
- [ ] `cargo clippy --workspace --all-targets --all-features -- -D warnings`
- [ ] `cargo test --workspace --locked`
- [ ] `cargo test --workspace --all-features --locked`
- [ ] Relevant architecture / dep-graph / docs checks for touched areas
- [x] Additional scenario, benchmark, or manual checks when behavior changed
- [ ] If this changes config/env fallback, limits, or defaults: include before/after behavior and regression coverage for explicit path, fallback path, and boundary values
- [x] If tests mutate process-global env: document how state is restored or serialized

Commands and evidence:

```text
cargo fmt --all -- --check
PASS

cargo test -p loongclaw-daemon --test integration migrate_cli_ -- --nocapture
PASS
- 7 migrate-related integration tests passed
- includes new CLI parse coverage and existing migrate behavior tests

cargo test -p loongclaw-daemon --test integration cli_migrate_help_explains_explicit_migration_flow -- --nocapture
PASS
- migrate subcommand help text validated

cargo run -p loongclaw-daemon -- migrate --help
PASS
- help renders correctly and shows --mode plus all supported values

cargo run -p loongclaw-daemon -- migrate --mode discover --input .
PASS
- command runs successfully without panic
- discovery completed and reported one detected source in the repo root

cargo clippy -p loongclaw-daemon --all-targets --all-features -- -D warnings
PASS

cargo clippy --workspace --all-targets --all-features -- -D warnings
NOT PASSING CURRENTLY
- failed in unrelated existing app test code:
  - crates/app/src/acp/acpx.rs (missing AtomicUsize / Ordering imports in tests)
  - crates/app/src/feishu/token_store.rs (unused import in tests)
- these failures are outside this PR’s touched files


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a top-level "migrate" command to manage legacy migrations with required --mode, optional --input/--output, source hints, source-id/primary-source-id, --safe-profile-merge, --apply-external-skills-plan, --force, and a --json output toggle. Outputs can be JSON or human-readable summaries.

* **Documentation**
  * Enhanced migrate help text explaining the power-user migration flow and required flags.

* **Tests**
  * Added integration tests for CLI parsing, help text, required-mode behavior, UX error messages, and policy-enforced path checks.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

### CODE REVIEW

File: crates/daemon/src/main.rs [hunk 1]
```diff
@@ -188,6 +188,29 @@ async fn main() {
...
+            output,
+            source,
+            mode,
+            json,
+            source_id,
+            safe_profile_merge,
+            primary_source_id,
+            apply_external_skills_plan,
+            force,
+        }),
```
[the remark this answers is not in the document]
Comment (gh-xj) [reply crates/daemon/src/main.rs:213]: Fixed in `fdde89748058a8de262a996ab1cc8605751f268e`.

`Commands::Migrate` still reaches `migrate_cli::run_migrate_cli`, but that function no longer executes migration logic directly. It now:
- bootstraps a kernel context with `mvp::context::bootstrap_kernel_context_with_config(...)`;
- dispatches `ToolCoreRequest { tool_name: "claw.migrate", ... }` through `mvp::tools::execute_tool(...)`; and
- therefore runs the migrate path under the same kernel capability/policy/audit enforcement as the other tool-mediated paths.

The shared kernel bootstrap in `crates/app/src/context.rs` was also updated to derive tool runtime + policy extensions from the supplied LoongClaw config instead of the global default, so explicit `tools.file_root` policy is enforced on this path as well.

Behavior was reverified on the pushed SHA with:
- `cargo fmt --all -- --check`
- `cargo clippy -p loongclaw-daemon --all-targets --all-features -- -D warnings`
- `cargo test -p loongclaw-daemon --test integration migrate_cli_ -- --nocapture`
- `cargo test -p loongclaw-daemon --all-features --test integration run_migrate_cli_ -- --nocapture`
- `cargo run -p loongclaw-daemon -- migrate --help`
- `cargo run -p loongclaw-daemon -- migrate --mode discover --input .`

The regression `run_migrate_cli_apply_mode_rejects_output_path_outside_configured_file_root` was added to prove writes outside an explicit configured file root are denied.

**Comment by gh-xj** [comment]
Follow-up on top of the kernel-routing fix is now on PR head `d7a0feffdfe300dd5bc83bc86412babcabb2305a`.

This UX pass does two things for `loongclaw migrate`:
- translates raw tool validation failures like `payload.output_path` into CLI-facing flag guidance such as ``--output` is required for `loongclaw migrate --mode apply```;
- adds a mode quick reference to `loongclaw migrate --help` so required flags are visible before users hit the error path.

Verified on this head with:
- `cargo fmt --all -- --check`
- `cargo clippy -p loongclaw-daemon --all-targets --all-features -- -D warnings`
- `cargo test -p loongclaw-daemon --test integration migrate_cli_ -- --nocapture`
- `cargo run -p loongclaw-daemon -- migrate --help`

**Review by gh-xj — APPROVED** [review]
